lntest/itest/lnd_test.go

```diff
@@ -578,6 +578,15 @@ func completePaymentRequests(ctx context.Context, client lnrpc.LightningClient,
 			return false
 		}
 
+		// If the number of open channels is now lower than before
+		// attempting the payments, it means one of the payments
+		// triggered a force closure (for example, due to an incorrect
+		// preimage). Return early since it's clear the payment was
+		// attempted.
+		if len(newListResp.Channels) < len(listResp.Channels) {
+			return true
```

Comment:
Nice catch. In this context, I think the decrease in the number of open channels indicates there's something wrong with the payment. So, implicitly, the payment was attempted.

Unrelated to this PR, the function `completePaymentRequests` has a very limited scope. Not only it cannot be run in parallel, but also requires a single change to a single channel for a single node at a time. It would be much better if it's more specific, say, checking a specific channel using the channel point.

Reply:
> It would be much better if it's more specific, say, checking a specific channel using the channel point.

Agree completely. Though, given how extensively `completePaymentRequests` is currently used in the itests, that would be a more significant undertaking...